src/mock_crypto.rs

```diff
@@ -71,6 +71,13 @@ pub(crate) mod crypto_impl {
                      Serialize)]
             pub(crate) struct PublicKey(pub(crate) [u8; PUBLICKEYBYTES]);
 
+            impl PublicKey {
+                // TODO: just to make it build
+                pub(crate) fn from_slice(_bs: &[u8]) -> Option<Self> {
```

Comment:
hm, do we really need `from_slice()`, if we have `from_bytes()`? :)

Reply:
I think these from_slice functions are mocking rust_sodium though?

Reply:
Yeah, you're right. I missed the file name, sorry :)